x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx

```diff
@@ -101,6 +102,11 @@ export function LensEditConfigurationFlyout({
       const adaptersTables = previousAdapters.current?.tables?.tables as Record<string, Datatable>;
       const [table] = Object.values(adaptersTables || {});
       if (table) {
+        // there are cases where a query can return a big amount of columns
+        // at this case we don't suggest all columns in a table but the first
+        // MAX_NUM_OF_COLUMNS
+        const columns = Object.keys(table.rows?.[0]) ?? [];
```

Comment:
why not reading from `table.columns`?

Reply:
It has only the used columns and not everything

Reply:
It can be still subject to sampling issues i.e. some column might have a `null` value on the first entry.
Anyway, it is ok to start with I guess and in case we'll change it if any bug arise.

Reply:
It is just to display a warning in case the user has a query such as `from logs` where all the columns are returned and we only render the first 5 in the table. It will be fine.